test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedAppWithRid.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.Assertions;
+using System.IO;
+using Xunit;
+using static Microsoft.NET.TestFramework.Commands.MSBuildTest;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildASelfContainedAppWithRid : SdkTest
+    {
+        [Fact]
+        public void It_builds_a_runnable_output()
```

Comment:
You can pass in a value to `CopyTestAsset` which it will use for the folder path, so you can have a long and descriptive test method name, but the path on disk will still be short enough.  Here's an example: https://github.com/dotnet/sdk/blob/b11de8deb43adec64e53fc441b73799b979a78f7/test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs#L526